File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/interactors/CardFilterInteractorImpl.java
package com.dbottillo.mtgsearchfree.interactors;

import com.dbottillo.mtgsearchfree.model.CardFilter;
import com.dbottillo.mtgsearchfree.model.storage.CardsPreferences;
import com.dbottillo.mtgsearchfree.util.Logger;

import io.reactivex.Observable;

public class CardFilterInteractorImpl implements CardFilterInteractor {

    private final CardsPreferences cardsPreferences;
    private final Logger logger;

    public CardFilterInteractorImpl(CardsPreferences cardsPreferences, Logger logger) {
        this.logger = logger;
        this.cardsPreferences = cardsPreferences;
        logger.d("created");
    }

    public Observable<CardFilter> load() {
        logger.d("loadSet");
        return Observable.just(cardsPreferences.load());
    }

    public void sync(CardFilter filter) {
        logger.d("sync");
        cardsPreferences.sync(filter);
    }

}

File: MTGSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/interactors/CardFilterInteractorImpl.kt
package com.dbottillo.mtgsearchfree.interactors

import com.dbottillo.mtgsearchfree.model.CardFilter
import com.dbottillo.mtgsearchfree.storage.CardsPreferences
import com.dbottillo.mtgsearchfree.util.Logger

import io.reactivex.Observable

class CardFilterInteractorImpl(
    private val cardsPreferences: CardsPreferences,
    private val logger: Logger
) : CardFilterInteractor {

    init {
        logger.d("created")
    }

    override fun load(): Observable<CardFilter> {
        logger.d("loadSet")
        return Observable.just(cardsPreferences.load())
    }

    override fun sync(filter: CardFilter) {
        logger.d("sync")
        cardsPreferences.sync(filter)
    }
}
